mindmeld/components/question_answerer.py

```diff
@@ -451,803 +528,2178 @@ def _generate_mapping_data(embedder_model, embedding_fields):
+            if field:
+                if field in self._filter_queries:
+                    msg = f"Found a duplicate filter clause against `{field}` field name. " \
+                          "Utilizing only latest input."
+                    logger.warning(msg)
+                self._filter_queries.update(
+                    {field: {"gt": gt, "gte": gte, "lt": lt, "lte": lte, "et": et,
+                             "boolean": boolean}}
                 )
 
-            return clause, functions
+            # filter that operates on strings; extract field name and query strings then
+            else:
+                for key, filter_text in kwargs.items():
+                    if key in self._filter_queries:
+                        msg = f"Found a duplicate filter clause against `{key}` field name. " \
+                              "Utilizing only latest input."
+                        logger.warning(msg)
+                    self._filter_queries.update({key: {"filter_text": filter_text}})
+
+            return self
+
+        def sort(self, field, sort_type=None, location=None):
+
+            if field in self._sort_queries:
+                msg = f"Found a duplicate sort clause against `{field}` field name. " \
+                      "Utilizing only latest input."
+                logger.warning(msg)
+            self._sort_queries.update(
+                {field: {"sort_type": sort_type, "location": location}})
+
+            return self
+
+        def execute(self, size=10):
+
+            try:
+                index_resources = NonElasticsearchQuestionAnswerer.ALL_INDICES.get(self.index_name)
+                index_all_ids = (
+                    NonElasticsearchQuestionAnswerer.ALL_INDICES.get_all_ids(self.index_name)
+                )
+            except KeyError:
+                msg = f"The index `{self.index_name}` looks unavailable. " \
+                      f"Consider running `.load_kb(...)` to create indices " \
+                      f"before search/filter/sort queries. "
+                logger.error(msg)
+                return []
+
+            # get results (aka. curated_docs) for `query` clauses, in decreasing order of similarity
+            n_scores = 0
+            scores = {}
+            for field_name, kwargs in self._search_queries.items():
+                query_type = kwargs["query_type"]
+                value = kwargs["value"]
+                field_resource = index_resources.get(field_name, None)
+                if not field_resource:
+                    msg = f"The field `{field_name}` is not available in index `{self.index_name}`."
+                    logger.error(msg)
+                    raise ValueError("Invalid knowledge base field '{}'".format(field_name))
+                this_field_scores = field_resource.do_search(query_type, value)
+                scores = {_id: (scores.get(_id, 0.0) + _score) / (n_scores + 1)
```

Comment:
Seems like a Counter dict collection might be more intuitive here

Reply:
Just to confirm, did you suggest to use [Counter](https://docs.python.org/3/library/collections.html#collections.Counter). If so, it does not support floats as values i think. let me know if I got it wrong.